### PR TITLE
Add an Export button for Devices list

### LINKED ISSUE #1185: Feature: Add an Import and an Export button for Devices list
**Describe the solution you'd like**
It could be interesting to have a button Import and Export to get and publish easily every Devices in the Device List.

- [ ] Import/Export buttons must be on Listing Devices view
- [ ] Import should override existing data

### PR DESCRIPTION
**Pull request by audserraCGI**
## Description

What's new?

- Closes #1185

## What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Tests
- [ ] Other

### CODE REVIEW

File: src/AzureIoTHub.Portal.Server/Services/LoRaWanDeviceService.cs [hunk 1]
```diff
@@ -131,5 +133,18 @@ protected override async Task DeleteDeviceInDatabase(string deviceId)
                 throw new InternalServerErrorException($"Unable to delete the LoRaWAN device {deviceId}", e);
             }
         }
+
+        public override async Task<Stream> ExportDeviceList()
+        {
+            Console.WriteLine("Call to ExportDeviceList() on LoRaWanDeviceService");
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal.Server/Services/LoRaWanDeviceService.cs:139]: Use the ILogger

File: src/docker-compose.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ version: '3.4'
 
 services:
   database:
-    image: postgres:14-alpine
+    image: postgres:15-alpine
```
Comment (kbeaugrand) [inline src/docker-compose.yml:5]: remove that from this PR

File: src/AzureIoTHub.Portal.Server/Services/LoRaWanDeviceService.cs [hunk 3]
```diff
@@ -131,5 +133,18 @@ protected override async Task DeleteDeviceInDatabase(string deviceId)
...
+
+            var query = portalDbContext.LorawanDevices
+                .Include(device => device.Tags);
+
+            //var test = (await query.ToListAsync()).ToCsv();
+            //Console.WriteLine(test);
+
+            return new FileStream("test.csv", FileMode.OpenOrCreate);
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal.Server/Services/LoRaWanDeviceService.cs:147]: Generate a dynamic name (e.g: based on date...)

**Review by kbeaugrand — COMMENTED** [review]
Hi, 

Globally 2 things: 
- I think this might be only an option in an Admin menu instead of the device list page...
- The best for us might to use the Device registry as the source to export regardless of the device kind

**Review by hocinehacherouf — APPROVED** [review]

**Pull request merged**